upkie/cpp/actuation/RobotSimulator.cpp

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2024 Inria
+
+#include "upkie/cpp/actuation/RobotSimulator.h"
+
+#include <iostream>
+
+bool RobotSimulatorClientAPI::changeDynamics(
+    int bodyUniqueId, int linkIndex,
+    struct RobotSimulatorChangeDynamicsArgs& args) {
+  b3PhysicsClientHandle sm = m_data->m_physicsClientHandle;
+  if (sm == 0) {
+    b3Warning("Not connected to physics server.");
+    return false;
+  }
+  b3SharedMemoryCommandHandle command = b3InitChangeDynamicsInfo(sm);
+  b3SharedMemoryStatusHandle statusHandle;
+
+  if (args.m_activationState >= 0) {
+    b3ChangeDynamicsInfoSetActivationState(command, bodyUniqueId,
+                                           args.m_activationState);
+  }
+  if (args.m_mass >= 0) {
+    b3ChangeDynamicsInfoSetMass(command, bodyUniqueId, linkIndex, args.m_mass);
+  }
+  if (!(args.m_localInertiaDiagonal[0] == 0 &&
+        args.m_localInertiaDiagonal[1] == 0 &&
+        args.m_localInertiaDiagonal[2] == 0)) {
+    b3ChangeDynamicsInfoSetLocalInertiaDiagonal(
+        command, bodyUniqueId, linkIndex, args.m_localInertiaDiagonal);
+  }
+  if (args.m_lateralFriction >= 0) {
+    b3ChangeDynamicsInfoSetLateralFriction(command, bodyUniqueId, linkIndex,
+                                           args.m_lateralFriction);
+  }
+
+  if (args.m_spinningFriction >= 0) {
```

Comment:
nit: here what will happen is

1. The compiler casts `0` to `0.0`
2. Comparison is performed

This is not incorrect. The nit is:

```suggestion
  if (args.m_spinningFriction >= 0.0) {
```